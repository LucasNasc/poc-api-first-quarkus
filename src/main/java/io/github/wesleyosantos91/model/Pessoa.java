package io.github.wesleyosantos91.model;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
import io.swagger.model.RequestPostPessoa;
import io.swagger.model.RequestPutPessoa;
import io.swagger.model.ResponsePessoa;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import java.io.Serializable;
import java.time.LocalDate;

@Builder
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "pessoa")
public class Pessoa extends PanacheEntityBase implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "codigo")
    private Long codigo;

    @Column(name = "nome")
    private String nome;

    @Column(name = "data_nascimento")
    private LocalDate dataNascimento;

    @Column(name = "cpf")
    private String cpf;

    @Column(name = "email")
    private String email;

    public static Pessoa requestPostPessoatoEntity(RequestPostPessoa requestPostPessoa) {
        return builder()
                .nome(requestPostPessoa.getNome())
                .dataNascimento(requestPostPessoa.getDataNascimento())
                .cpf(requestPostPessoa.getCpf())
                .email(requestPostPessoa.getEmail())
                .build();
    }
    public static void requestPutPessoatoEntity(RequestPutPessoa requestPutPessoa, Pessoa pessoa) {
        pessoa.setNome(requestPutPessoa.getNome());
        pessoa.setEmail(requestPutPessoa.getEmail());
        pessoa.setDataNascimento(requestPutPessoa.getDataNascimento());
    }

    public static ResponsePessoa pessoaToResponsePessoa(Pessoa pessoa) {

        ResponsePessoa responsePessoa = new ResponsePessoa();
        responsePessoa.setCodigo(pessoa.getCodigo());
        responsePessoa.setNome(pessoa.getNome());
        responsePessoa.setCpf(pessoa.getCpf());
        responsePessoa.setDataNascimento(pessoa.getDataNascimento());
        responsePessoa.setEmail(pessoa.getEmail());

        return responsePessoa;
    }
}