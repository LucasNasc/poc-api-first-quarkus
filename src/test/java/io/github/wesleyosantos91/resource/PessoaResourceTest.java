package io.github.wesleyosantos91.resource;

import io.github.wesleyosantos91.MysqlTestLifecycleManager;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.http.ContentType;
import io.swagger.model.RequestPostPessoa;
import io.swagger.model.RequestPutPessoa;
import org.apache.http.HttpStatus;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.MethodOrderer;
import org.junit.jupiter.api.Order;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestMethodOrder;

import javax.transaction.Transactional;
import java.time.LocalDate;

import static io.restassured.RestAssured.given;
import static org.hamcrest.CoreMatchers.is;

@QuarkusTest
@QuarkusTestResource(MysqlTestLifecycleManager.class)
@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
@Transactional
class PessoaResourceTest {

    private RequestPostPessoa requestPostPessoa;

    private RequestPutPessoa requestPutPessoa;

    @BeforeEach
    void setUp() throws Exception {

        requestPostPessoa = new RequestPostPessoa();
        requestPostPessoa.setNome("Gabriel Ducati Teixeira");
        requestPostPessoa.setCpf("10275977340");
        requestPostPessoa.setDataNascimento(LocalDate.of(1995, 10, 10));
        requestPostPessoa.setEmail("gabriel.teixeira@uol.com.br");

        requestPutPessoa = new RequestPutPessoa();
        requestPutPessoa.setNome("Guilherme Machado Gomes");
        requestPutPessoa.setDataNascimento(LocalDate.of(1972, 9, 12));
        requestPutPessoa.setEmail("guilherme.gomes@uol.com.br");
    }

    @Test
    @DisplayName("Deve retorna uma pessoa pelo seu código")
    @Order(1)
    void consultarPeloCodigo() {
        String resultado = given()
                .pathParam("codigo", 1L)
                .get("/pessoas/{codigo}")
                .then()
                .statusCode(HttpStatus.SC_OK)
                .extract().asString();
        Approvals.verifyJson(resultado);

    }

    @Test
    @DisplayName("Deve lançar ObjectNotFoundException")
    @Order(2)
    void consultarPeloCodigoException() {
        given()
                .pathParam("codigo", 2L)
                .get("/pessoas/{codigo}")
                .then()
                .statusCode(HttpStatus.SC_NOT_FOUND);

    }

    @Test
    @DisplayName("deve retornar todos umas lista de pessoas")
    @Order(3)
    void consultarTodos() {
        given()
                .get("/pessoas")
                .then()
                .statusCode(HttpStatus.SC_OK)
                .body("size()", is(1));
    }

    @Test
    @DisplayName("Deve alterar a o objeto pessoa")
    @Order(4)
    void alterarPessoa() {
        String resultado = given()
                .pathParam("codigo", 1L)
                .request()
                .header("Accept", ContentType.ANY)
                .header("Content-type", ContentType.JSON)
                .body(requestPutPessoa)
                .when()
                .put("/pessoas/{codigo}")
                .then()
                .statusCode(HttpStatus.SC_OK)
                .extract().asString();
        Approvals.verifyJson(resultado);

    }

    @Test
    @DisplayName("Deve cadastrar uma pessoa")
    @Order(5)
    void cadastrarPessoa() {
        String resultado = given()
                .request()
                .header("Accept", ContentType.ANY)
                .header("Content-type", ContentType.JSON)
                .body(requestPostPessoa)
                .when()
                .post("/pessoas")
                .then()
                .statusCode(HttpStatus.SC_CREATED)
                .extract().asString();

        Approvals.verifyJson(resultado);


    }

    @Test
    @DisplayName("Deve excluir uma pessoa pelo seu código")
    @Order(6)
    void excluirPessoa() {
        given()
                .pathParam("codigo", 1L)
                .request()
                .header("Accept", ContentType.ANY)
                .header("Content-type", ContentType.JSON)
                .body(requestPostPessoa)
                .when()
                .delete("/pessoas/{codigo}")
                .then()
                .statusCode(HttpStatus.SC_NO_CONTENT);
    }

    @Test
    @DisplayName("eve lançar ObjectNotFoundException ao tentar excluir uma pessoa pelo seu código")
    @Order(7)
    void excluirPessoaException() {
        given()
                .pathParam("codigo", 1L)
                .request()
                .header("Accept", ContentType.ANY)
                .header("Content-type", ContentType.JSON)
                .body(requestPostPessoa)
                .when()
                .delete("/pessoas/{codigo}")
                .then()
                .statusCode(HttpStatus.SC_NOT_FOUND);
    }

    @Test
    @DisplayName("Deve lançar ObjectNotFoundException ao tentar alterar a o objeto pessoa")
    @Order(8)
    void alterarPessoaException() {
        given()
                .pathParam("codigo", 1L)
                .request()
                .header("Accept", ContentType.ANY)
                .header("Content-type", ContentType.JSON)
                .body(requestPutPessoa)
                .when()
                .put("/pessoas/{codigo}")
                .then()
                .statusCode(HttpStatus.SC_NOT_FOUND);

    }
}