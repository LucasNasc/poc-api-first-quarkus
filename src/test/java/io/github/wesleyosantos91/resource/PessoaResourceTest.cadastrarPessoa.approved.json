{
  "codigo": 2,
  "cpf": "10275977340",
  "email": "gabriel.teixeira@uol.com.br",
  "nome": "Gabriel Ducati Teixeira"
}